src/form/Form.tsx

```diff
@@ -779,8 +805,9 @@ function Form({
   }, [steps]);
 
   useEffect(() => {
-    if (stepKey) getNewStep(stepKey);
-  }, [stepKey]);
+    if (stepKey && (prevStepKey !== stepKey || prevAuthId !== initState.authId))
```

Comment:
why do you need to do these checks? doesn't this function being triggered imply one of these is different?